experiments/AMIP/moist_mpi_earth/push_pull.jl

```diff
@@ -1,180 +1,152 @@
-T_S = ClimaCore.Fields.zeros(boundary_space) # temperature
-z0m_S = ClimaCore.Fields.zeros(boundary_space)
-z0b_S = ClimaCore.Fields.zeros(boundary_space)
-ρ_sfc = ClimaCore.Fields.zeros(boundary_space)
-q_sfc = ClimaCore.Fields.zeros(boundary_space)
-albedo = ClimaCore.Fields.zeros(boundary_space)
-
-F_A = ClimaCore.Fields.zeros(boundary_space) # aerodynamic turbulent fluxes
-F_E = ClimaCore.Fields.zeros(boundary_space) # aerodynamic turbulent fluxes
-F_R = ClimaCore.Fields.zeros(boundary_space) # radiative fluxes
-P_liq = ClimaCore.Fields.zeros(boundary_space) # m/s precip
 """
-   atmos_push!(atmos_sim, boundary_space, F_A, F_R, F_E, P_liq, parsed_args)
-
+   atmos_push!(cs)
 updates F_A, F_R, P_liq, and F_E in place based on values used in the atmos_sim for the current step.
 """
-function atmos_push!(atmos_sim, boundary_space, F_A, F_R, F_E, P_liq, parsed_args)
-    F_A .= ClimaCore.Fields.zeros(boundary_space)
-    dummmy_remap!(F_A, atmos_sim.integrator.p.dif_flux_energy)
-    F_E .= ClimaCore.Fields.zeros(boundary_space)
-    dummmy_remap!(F_E, atmos_sim.integrator.p.dif_flux_ρq_tot)
-    P_liq .= ClimaCore.Fields.zeros(boundary_space)
-    dummmy_remap!(P_liq, atmos_sim.integrator.p.col_integrated_precip)
-    F_R .= ClimaCore.Fields.zeros(boundary_space)
-    parsed_args["rad"] == "gray" ? dummmy_remap!(F_R, level(atmos_sim.integrator.p.ᶠradiation_flux, half)) : nothing
+function atmos_push!(cs)
+    atmos_sim = cs.model_sims.atm
+    csf = cs.fields
+    dummmy_remap!(csf.F_A, atmos_sim.integrator.p.dif_flux_energy)
+    dummmy_remap!(csf.F_E, atmos_sim.integrator.p.dif_flux_ρq_tot)
+    dummmy_remap!(csf.P_liq, atmos_sim.integrator.p.col_integrated_precip)
+    cs.parsed_args["rad"] == "gray" ? dummmy_remap!(csf.F_R, level(atmos_sim.integrator.p.ᶠradiation_flux, half)) :
+    nothing
 end
 
 """
-   land_pull!(slab_sim::SlabSimulation, F_A, F_R, _...)
-
+   land_pull!(cs)
 Updates the slab_sim cache state in place with the current values of F_A and F_R.
 """
-function land_pull!(slab_sim::SlabSimulation, F_A, F_R, _...)
-    @. slab_sim.integrator.p.F_aero = F_A
-    @. slab_sim.integrator.p.F_rad = F_R
+function land_pull!(cs)
+    slab_sim = cs.model_sims.lnd
+    csf = cs.fields
+    @. slab_sim.integrator.p.F_aero = csf.F_A
+    @. slab_sim.integrator.p.F_rad = csf.F_R
 end
 
 """
-   land_pull!(slab_sim::BucketSimulation, F_A, F_R, F_E, P_liq, ρ_sfc)
-
+   land_pull!(cs)
 Updates the slab_sim cache state in place with the current values of F_A, F_R, F_E, P_liq, and ρ_sfc.
-
 The surface air density is computed using the atmospheric state at the first level and making ideal gas
 and hydrostatic balance assumptions. The land model does not compute the surface air density so this is
 a reasonable stand-in.
 """
-function land_pull!(slab_sim::BucketSimulation, F_A, F_R, F_E, P_liq, ρ_sfc)
-    FT = eltype(F_A)
-    @. slab_sim.integrator.p.bucket.ρ_sfc = ρ_sfc
-    @. slab_sim.integrator.p.bucket.SHF = F_A
+function land_pull!(cs)
```

Comment:
How do we execute the different methods of land_pull? prior to this, they dispatched on the type of land model.
I think we need land_pull!(land_sim::BucketSimulation, cs) and land_pull!(land_sim::SlabSimulation, cs). in general we will have to dispatch on the type of model, so this format might be nice for all (land, ocean, ice)?

Reply:
Yeah, this was just a temporary solution for this AMIP case, until we deploy Ben's `Simulations` interface, which will def allow this multiple dispatch - I should have been clearer in the PR description above. Are you guys happy with this for now, since ` land_pull.jl` will be scrapped eventually anyway?

Reply:
Hm, not sure I understand. If we leave as is, what method of land_pull! will be executed if the land model is a bucket, vs the land modeling being a slab? Alternatively, we can scrap the slab being an option for the land?

Reply:
(having a stand-in is fine, but my point was more to make sure the temporary soln has the capabilities we want to support)

Reply:
No, it is a good point! From the AMIP roadmap perspective, I don't think we need to support a thermal-slab land, but if you think it would be valuable to have it here, then I can add it. Lmk when you're free next, maybe we can chat on Zoom and decide 😎 .

Reply:
free anytime this morning! I'm fine losing the thermal slab land in this experiment directory though 🌤️